src/main/webapp/WEB-INF/view/activity.jsp

```diff
@@ -10,19 +18,61 @@ List<Activity> activities = (List<Activity>) request.getAttribute("activities");
 <head>
   <title>ActivityFeed</title>
   <link rel="stylesheet" href="/css/main.css">
+
+  <style>
+    #activityfeed {
+      background-color: white;
+      height: 500px;
+      overflow-y: scroll
+    }
+  </style>
+
+  <script>
+    // scroll the activityfeed div to the bottom
+    function scrollChat() {
+      var chatDiv = document.getElementById('chat');
+      chatDiv.scrollTop = chatDiv.scrollHeight;
+    };
+  </script>
+
 </head>
-<body>
+<body onload="scrollChat()">
   <%@ include file = "/navigations.jsp" %>
 
-  <div id="container">
-    <h2 style="color:blue; text-aligned:left; margin-bottom:25px;"> ACTIVITY </h2>
-    <h3> Here's everything that's happened on the site so far! </h3>
-
-    <ul class="mdl-list">
-      <% for (Activity activity : activities) { %>
-        <li> <%= activity.getAction().name() %> </li>
-      <% } %>
-    </ul>
-  </div>
+    <div id="container">
+      <h2 style="color:blue; text-aligned:left; margin-bottom:25px"> ACTIVITY </h2>
+      <h3> Here's everything that's happened on the site so far! </h3>
+        <div id="activityfeed">
+          <ul>
+            <%
+              for (int i = activities.size() -1; i>-1; i--) {
+                Activity activity = activities.get(i);
+                if(activity.isPublic()) {
+                  Action action = activity.getAction();
+                  UUID id = activity.getId();
+                  UUID idOwnerId = activity.getOwnerId();
+                  String time = Util.FormatDateTime(activity.getCreationTime());
+                  if(action == Action.REGISTER_USER){
```

Comment:
You should be able to use a `switch` here if you find that cleaner.

Reply:
Yep, it is way cleaner, especially since we can increase the number of actions that can take place. Thanks!